include/cuco/detail/pair.cuh

```diff
@@ -133,6 +133,8 @@ using pair_type = cuco::pair<K, V>;
 template <typename F, typename S>
 __host__ __device__ pair_type<F, S> make_pair(F&& f, S&& s) noexcept
 {
-  return pair_type<F, S>{std::forward<F>(f), std::forward<S>(s)};
+  using decayed_F = std::decay_t<F>;
+  using decayed_S = std::decay_t<S>;
+  return pair_type<decayed_F, decayed_S>{std::forward<decayed_F>(f), std::forward<decayed_S>(s)};
```

Comment:
Can we use `move` here instead?

Reply:
`std::forward` will have the same semantics as `std::move` if `F` or `S` are r-value references. 

You can think of `std::forward` as a more generic form of `std::move`.

Reply:
Will there be a return type mismatch with the function signature?

Reply:
Great point! Used `auto` return type instead.

Reply:
@jrhemstad yeah, I guess my point is that since `decayed_F` is not a reference type, then `forward<decayed_F>` will always return an r-value, so it's identical to just using `move` in this case - is my understanding correct here? and so I was thinking that `move` is just shorter to use here.

Reply:
I suppose you could use `std::forward` with the original type as well